### ISSUE TITLE
[HOLD App#11795] [$1000] Workspace - Invite new members list always scrolls to top after selecting 4 or more

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#12998]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
  1. Click on avatar
  2. Click on Workspace
  3. Select Workspace
  4. Click Manage Members
  5. Click on Invite Members
  6. Select at least 8 Members

## Expected Result: 
User should be able to view the most recent selected member

## Actual Result: 
List scrolls to top with first selected member

## Workaround:
unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.31-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/203805699-c6c3a8fd-717d-4558-a827-a8f92140c9ec.mp4

https://user-images.githubusercontent.com/43996225/203805670-d0d54cfe-d7c9-4445-904f-d08c165fdea2.mov

https://user-images.githubusercontent.com/43996225/203805688-b4d64210-d810-4cc7-b205-11a6923b5174.mov

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @syedsaroshfarrukhdot 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1669292832144069

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0174a96dd459e1553d</li>
        <li>Upwork Job ID: 1598666700904955904</li>
        <li>Last Price Increase: 2022-12-08</li>
    </ul>
</details>

**Comment by hungvu193** [comment]
I think this a feature, not a bug.
https://github.com/Expensify/App/blob/39b2f9924060c7381b2ff37563a5af88360b0d9d/src/components/OptionsSelector/BaseOptionsSelector.js#L110-L114
In case we don't wanna it scroll to top (which's kind of annoying with me too) we can comment or delete this piece of code.
```diff
-            // If we just toggled an option on a multi-selection page, scroll to top
-           // if (this.props.selectedOptions.length !== prevProps.selectedOptions.length) {
-            //     this.scrollToIndex(0);
-            //     return;
-           // }
```

**Comment by davidcardoza** [comment]
Hmm...This feels like a bug, but I want a second opinion before moving forward with creating a job. Applying the engineering label.

**Comment by cristipaval** [comment]
Hey @davidcardoza ! Yes, this is reproducible and I would also consider it a bug.

**Comment by davidcardoza** [comment]
Job posted - https://www.upwork.com/jobs/~01770333bb615437f2 waiting for proposals

**Comment by syedsaroshfarrukhdot** [comment]
Proposal:-

We can fix this issue by pushing most recently selected user to top

**Comment by cristipaval** [comment]
@davidcardoza How did you create the upwork job? I think you should have applied the External label to this issue and everything would have been created automatically along with some others like C+ assignment.

**Comment by cristipaval** [comment]
Ok, I removed the Upwork job created by you @davidcardoza and applied `External` label. I had to remove it to avoid a duplicate job posting on Upwork.

**Comment by amyevans** [comment]
Hmm I _think_ as @hungvu193 stated this behavior was implemented purposefully, cc @roryabraham

**Comment by syedsaroshfarrukhdot** [comment]
@amyevans Yes, I it might be implemented purposefully but it effects user as last selected user is not visible when it scrolled to top with the first selected user at top. So we can use My Proposal (https://github.com/Expensify/App/issues/12998#issuecomment-1334567889) to push latest selected user to top to avoid the issue.

**Comment by cristipaval** [comment]
Hmm, looks like it's not clear if this is a bug or something implemented on purpose. Opened a [Slack convo](https://expensify.slack.com/archives/C01GTK53T8Q/p1670238226611139) about it.

**Comment by parasharrajat** [comment]
Thanks for the proposals. I will check it asap.

**Comment by parasharrajat** [comment]
This is being discussed on slack. I will chime in later. There are some instructions but I need to check what things will be affected by the expected changes. If there are things that also needs to be fixed with the new expected changes.

**Comment by cristipaval** [comment]
We're still discussing on Slack. I hope we get a conclusion asap.

**Comment by mountiny** [comment]
After discussing, I am putting this issue on hold for https://github.com/Expensify/App/issues/11795, there we will redesign the option lists so the UX is better and works fine across platforms. Better to address these issues which are not blockers in structured manner.

**Comment by cristipaval** [comment]
Thanks @mountiny ! I'll close the upwork job for now.

**Comment by syedsaroshfarrukhdot** [comment]
I am eligible for reporting bonus ?

**Comment by davidcardoza** [comment]
The issue was put on hold. Reporting bonuses are paid out only after an issue/bug has been resolved.

**Comment by cristipaval** [comment]
@JmillsExpensify , replying to [your comment](https://github.com/Expensify/App/issues/11795#issuecomment-1405801826) here.

We discussed about this issue [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1670238226611139). Please have a look when you've got a minute.

**Comment by JmillsExpensify** [comment]
Nice, thanks. I commented in the related Slack thread. Re-posing below.

> Looks like we never really decided what to do here. I personally like the sticky header and feel that it’s better than the auto-scroll up. That said, I think this is squarely new feature territory and I’m not sure this is really critical right now. I think I’d opt to keep punting this.

**Comment by davidcardoza** [comment]
I concur with the sentiment @JmillsExpensify,  the decision regarding the implementation of a sticky header versus auto-scroll functionality has yet to be determined and is probably deserving of a broader conversation to implement as a new feature. 

@cristipaval Is there any reason to keep this issue on hold?

**Comment by cristipaval** [comment]
We can disable the auto scroll functionality as a solution for this issue or just simply close this issue and disable the auto-scroll when we start a broader conversation about the sticky header.

**Comment by davidcardoza** [comment]
I vote the latter of the two option, let's punt until we have begun speccing out the sticky header.

**Comment by cristipaval** [comment]
I'm also leaning towards closing this one and disable the auto-scroll when we start working on the sticky header.

**Comment by cristipaval** [comment]
Closing for now, @JmillsExpensify feel free to reopen it if you don't agree.
